src/main/groovy/com/vherasymenko/avro/decoder/core/CourseInstallDecoderService.groovy

```diff
@@ -21,11 +21,29 @@ class CourseInstallDecoderService implements CourseInstallDecoderPort {
     }
 
     @Override
-    void decodeCourseInstallEvent( String event, int schemaId ) {
+    void decodeCourseInstallV1ToV1( String event, int schemaId ) {
+        def schema = new Schema.Parser().parse( registryClient.fetch( schemaId ) )
+        log.info( 'The avro schema for event decoding : ' + schema.toString( true ) )
+
+        decoder.decodeEventWithOldSchema( schema, event )
+    }
+
+    @Override
+    void decodeCourseInstallV1ToV2( String event, int schemaId ) {
         def writeSchema = new Schema.Parser().parse( registryClient.fetch( schemaId ) )
         def readSchema = new Schema.Parser().parse( new File( 'src/main/resources/avro/course_install_2.avsc' ) )
         log.info( 'The avro schema for event decoding : ' + readSchema.toString( true ) )
 
-        decoder.decodeEvent( writeSchema, readSchema, event )
+        decoder.decodeEventWithNewSchema( writeSchema, readSchema, event )
+    }
+
+    @Override
+    void decodeCourseInstallV1ToV3( String event, int schemaId ) {
+        def writeSchema = new Schema.Parser().parse( registryClient.fetch( schemaId ) )
+        def readSchema = new Schema.Parser().parse( new File( 'src/main/resources/avro/course_install_3.avsc' ) )
```

Comment:
I thought we were going to fetch **all** schema from the registry and not from the file system.  Am I wrong?